### PR TITLE
0.2.0-rc1 - Address PR comments for #37

### PR DESCRIPTION
**Pull request by valydia**
@marius-m made a great job on #37  - just addressing the final comments...

### CODE REVIEW

**Comment by valydia** [comment]
Hey folks! Is there any chance this PR could be reviewed / merged? I'd be happy to address any feedback on it 👍 ... cc @ben-j69

**Comment by ben-j69** [comment]
@valydia yes it is planned but we are very busy for the moment.

File: src/main/kotlin/com/klaxit/hiddensecrets/HiddenSecretsPlugin.kt [hunk 1]
```diff
@@ -153,12 +205,13 @@ open class HiddenSecretsPlugin : Plugin<Project> {
         }
 
         /**
-         * Copy Cpp files from the lib to the Android project if they don't exist yet
+         * Copy Cpp files from the lib to the Android project
+         * @param clean whether to overwrite existing files
          */
-        fun copyCppFiles() {
+        fun copyCppFiles(clean: Boolean) {
```
Comment (ben-j69) [inline src/main/kotlin/com/klaxit/hiddensecrets/HiddenSecretsPlugin.kt:211]: A parameter called `overwrite` would be clearer. And to not change too much the code and the actual behaviour a default value should be set. So the actual calls won't change
`        fun copyCppFiles(overwrite: Boolean = false) {}`

File: src/main/kotlin/com/klaxit/hiddensecrets/HiddenSecretsPlugin.kt [hunk 2]
```diff
@@ -168,23 +221,92 @@ open class HiddenSecretsPlugin : Plugin<Project> {
...
         /**
-         * Copy Kotlin file Secrets.kt from the lib to the Android project if it does not exist yet
+         * Copy Kotlin file Secrets.kt from the lib to the Android project
+         * @param clean whether to overwrite existing files
          */
-        fun copyKotlinFile() {
-            getKotlinFile()?.let {
-                println("$KOTLIN_FILE_NAME already exists")
-                return
+        fun copyKotlinFile(clean: Boolean) {
```
Comment (ben-j69) [inline src/main/kotlin/com/klaxit/hiddensecrets/HiddenSecretsPlugin.kt:227]: `Clean` is unclear here too.

File: src/main/kotlin/com/klaxit/hiddensecrets/HiddenSecretsPlugin.kt [hunk 3]
```diff
@@ -93,6 +100,42 @@ open class HiddenSecretsPlugin : Plugin<Project> {
...
+                        " Use: '${SAMPLE_FROM_PROPS}'")
+                }
+            } else {
+                throw IllegalArgumentException("Properties file is not defined!" +
+                    " Use: '${SAMPLE_FROM_PROPS}'")
+            }
+        }
+
+        /**
+         * Get properties file to hide secrets from
```
Comment (ben-j69) [inline src/main/kotlin/com/klaxit/hiddensecrets/HiddenSecretsPlugin.kt:123]: Get properties from the provided file

File: src/main/kotlin/com/klaxit/hiddensecrets/HiddenSecretsPlugin.kt [hunk 4]
```diff
@@ -236,69 +367,56 @@ open class HiddenSecretsPlugin : Plugin<Project> {
...
-                    error("Missing Kotlin file, please run gradle task : $TASK_COPY_KOTLIN")
-                }
-                //Resolve package name for C++ from the one used in Kotlin file
-                var kotlinPackage = Utils.getKotlinFilePackage(secretsKotlin)
-                if (kotlinPackage.isEmpty()) {
-                    println("Empty package in $KOTLIN_FILE_NAME")
-                    kotlinPackage = packageName
-                }
+                hideSecret(
+                    keyName = keyName,
```
Comment (ben-j69) [inline src/main/kotlin/com/klaxit/hiddensecrets/HiddenSecretsPlugin.kt:386]: Not need to repeat names and vars: 
`hideSecret(keyName, packageName, obfuscatedKey)`

File: src/main/kotlin/com/klaxit/hiddensecrets/HiddenSecretsPlugin.kt [hunk 5]
```diff
@@ -236,69 +367,56 @@ open class HiddenSecretsPlugin : Plugin<Project> {
...
+                copyKotlinFile(true)
+
+                val packageName = getPackageNameParam()
+                val propsFile = getPropertiesFile()
+                val props = getPropertiesFromFile(propsFile = propsFile)
+                println("Generating secrets from props: ${propsFile.path}")
+                props.entries.forEach { entry ->
+                    val keyName = entry.key as String
+                    val obfuscatedKey = Utils.encodeSecret(entry.value as String, packageName)
+                    hideSecret(
```
Comment (ben-j69) [inline src/main/kotlin/com/klaxit/hiddensecrets/HiddenSecretsPlugin.kt:414]: Not need to repeat names and vars: 
`hideSecret(keyName, packageName, obfuscatedKey)`

**Review by ben-j69 — CHANGES_REQUESTED** [review]
@valydia, thanks for the work. Can you rebase and adjust some last comments.

I have time allowed to close this improvement.

It will be merge and release in 0.2.0-rc1

**Comment by valydia** [comment]
Thanks for the review @ben-j69 ! I've addressed your comment in https://github.com/klaxit/hidden-secrets-gradle-plugin/pull/47/commits/a3cbe0462e136433f037b8295ef7c99541938bb7

**Comment by ben-j69** [comment]
Thanks @valydia , I will merge it in 0.2.0-rc1, test and adjust if necessary and then create a new release ;)

**Review by ben-j69 — APPROVED** [review]

**Comment by ben-j69** [comment]
Hello @valydia I made a mistake on rebase and I hade to manually copy the code to make a new commit here https://github.com/klaxit/hidden-secrets-gradle-plugin/pull/56 is it ok for you ? Not sure your name and @marius-m will appear on the repo.

If not can you create a new PR to master and I will recreate a 0.2.0

Thanks

**Comment by valydia** [comment]
Hi @ben-j69 ! Thanks for letting me know! You can proceed with your new commit! Cheers

**Pull request merged**